### PR TITLE
[FEATURE][MER-2707] Practice Activities Tab

### PR DESCRIPTION
**Pull request by simonchoxx**
[MER-2707](https://eliterate.atlassian.net/browse/MER-2707)

This PR adds the `Practice Activities` tab to the Instructor Dashboard Overview. 

Initially, a table is rendered to lists the pages that contain activities that are not found in surveys. 

In addition, a dropdown is added to be able to filter that list of pages by container.

When a user clicks on a page, the details of the activities contained in the selected page will be displayed (below the table of pages).

If the section has the `V2` analytics version, the results of the students' answers to the activity details for the `Multi Input` and `Likert` activity types are added. 

- The Multi Input activity type will show a table with the students' answers if the activity is of type "Text" or "Numeric", and will show the frequency of answers if the type is "Dropdown". 

- The Likert activity type will add a chart showing the results of the answers given by the students. 

https://github.com/Simon-Initiative/oli-torus/assets/16328384/1b112684-5f67-4125-a184-c19924cc2248

[MER-2707]: https://eliterate.atlassian.net/browse/MER-2707?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/oli_web/live/delivery/instructor_dashboard/instructor_dashboard_live.ex [hunk 1]
```diff
@@ -189,6 +189,40 @@ defmodule OliWeb.Delivery.InstructorDashboard.InstructorDashboardLive do
...
+        params: params,
+        view: :overview,
+        active_tab: :practice_activities
+      )
+      |> assign_new(:students, fn ->
+        Sections.enrolled_students(socket.assigns.section.slug)
+        |> Enum.reject(fn s -> s.user_role_id != 4 end)
+      end)
+      |> assign_new(:practice_activities, fn %{students: students} ->
+        Helpers.get_assessments(socket.assigns.section, students)
```
Comment (darrensiegel) [inline lib/oli_web/live/delivery/instructor_dashboard/instructor_dashboard_live.ex:209]: The `:practice_activities` assign is being populated with a call to `Helpers.get_assessments` which returns the **graded** pages.  Practice activities should be populated with the opposite set, those pages that are **not** graded.

File: lib/oli_web/components/delivery/practice_activities/practice_activities.ex [hunk 2]
```diff
@@ -0,0 +1,1012 @@
...
+    )
+    |> Repo.one()
+  end
+
+  defp get_activities(current_assessment, section, student_ids) do
+    activities =
+      from(aa in ActivityAttempt,
+        join: res_attempt in ResourceAttempt,
+        on: aa.resource_attempt_id == res_attempt.id,
+        where: res_attempt.lifecycle_state == :evaluated and aa.lifecycle_state == :evaluated,
```
Comment (darrensiegel) [inline lib/oli_web/components/delivery/practice_activities/practice_activities.ex:605]: The `res_attempt.lifecycle_state == :evaluated` clause must be removed.  Practice pages will usually never have their lifecycle_state transitioned into `:evaluated`

File: lib/oli_web/components/delivery/practice_activities/practice_assessments_table_model.ex [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        render_fn: &__MODULE__.render_order_column/3,
+        th_class: "pl-10"
+      },
+      %ColumnSpec{
+        name: :title,
+        label: "ASSESSMENT",
+        render_fn: &__MODULE__.render_assessment_column/3,
+        th_class: "pl-10"
+      },
+      %ColumnSpec{
```
Comment (darrensiegel) [inline lib/oli_web/components/delivery/practice_activities/practice_assessments_table_model.ex:22]: Practice pages will usually never have a Due Date.  This column should be removed.

**Review by darrensiegel — CHANGES_REQUESTED** [review]
Here is my first review:

1. I want to replace "ChartJS" with VegaLite.  VegaLite and the underlying Vega library position us better for the future development of new visualizations.

2. "Practice Activities" is loading only Graded pages, see my comment inline.  It needs to load all "practice pages". 

3. I do not think this implementation is going to work on courses that **only** contain pages.  Please test this with a course that simply contains practice pages at the root of the course. 

There are a few other comments inline.

**Comment by simonchoxx** [comment]
@darrensiegel About using VegaLite. Do you think it's ok to use VegaLite.js and build the chart from the React side or do you prefer to use the VegaLite library for elixir (https://hexdocs.pm/vega_lite/VegaLite.html#module-composing-graphics)?

**Review by darrensiegel — APPROVED** [review]

**Pull request merged**